cartography/intel/okta/factors.py

```diff
@@ -0,0 +1,140 @@
+
+    for current in okta_factor_list:
+        factors.append(transform_okta_user_factor(current))
+
+    return factors
+
+
+def transform_okta_user_factor(okta_factor_info):
+    """
+    Transform okta user factor into consumable data for the graph
+    :param okta_factor_info: okta factor information
+    :return: Dictionary of properties for the factor
+    """
+
+    # https://github.com/okta/okta-sdk-python/blob/master/okta/models/factor/Factor.py
+    factor_props = {}
+    factor_props["id"] = okta_factor_info.id
+    factor_props["factor_type"] = okta_factor_info.factorType
+    factor_props["provider"] = okta_factor_info.provider
+    factor_props["status"] = okta_factor_info.status
+    if okta_factor_info.created:
+        factor_props["created"] = okta_factor_info.created.strftime("%m/%d/%Y, %H:%M:%S")
+    else:
+        factor_props["created"] = None
+
+    if okta_factor_info.lastUpdated:
+        factor_props["okta_last_updated"] = okta_factor_info.lastUpdated.strftime("%m/%d/%Y, %H:%M:%S")
+    else:
+        factor_props["okta_last_updated"] = None
+
+    # we don't import Profile data into the graph due as it contains sensitive data
+    return factor_props
+
+
+def _load_user_factors(neo4j_session, user_id, factors, okta_update_tag):
+    """
+    Add user factors into the graph
+    :param neo4j_session: session with the Neo4j server
+    :param user_id: user to map factors to
+    :param factors: factors to add
+    :param okta_update_tag: The timestamp value to set our new Neo4j resources with
+    :return: Nothing
+    """
+
+    ingest = """
+    MATCH (user:OktaUser{id: {USER_ID}})
+    WITH user
+    UNWIND {FACTOR_LIST} as factor_data
+    MERGE (new_factor:OktaUserFactor{id: factor_data.id})
+    ON CREATE SET new_factor.firstseen = timestamp()
+    SET new_factor.factor_type = factor_data.factor_type,
+    new_factor.provider = factor_data.provider,
+    new_factor.status = factor_data.status,
+    new_factor.created = factor_data.created,
+    new_factor.okta_last_updated = factor_data.okta_last_updated,
+    new_factor.lastupdated = {okta_update_tag}
+    WITH user, new_factor
+    MERGE (user)-[r:FACTOR]->(new_factor)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {okta_update_tag}
+    """
+
+    neo4j_session.run(
+        ingest,
+        USER_ID=user_id,
+        FACTOR_LIST=factors,
+        okta_update_tag=okta_update_tag,
+    )
+
+
+def sync_users_factors(neo4j_session, okta_org_id, okta_update_tag, okta_api_key):
+    """
+    Sync user factors
+    :param neo4j_session: session with the Neo4j server
+    :param okta_org_id: okta organization id
+    :param okta_update_tag: The timestamp value to set our new Neo4j resources with
+    :param okta_api_key: Okta API key
+    :return: Nothing
+    """
+
+    logger.debug("Syncing Okta User Factors")
+
+    factor_client = _create_factor_client(okta_org_id, okta_api_key)
+    users = get_user_id_from_graph(neo4j_session, okta_org_id)
```

Comment:
As I mentioned in a previous pass of this PR, we already have the user IDs in memory here: https://github.com/lyft/cartography/blob/952620fdf3ff15c6d8a3872227ff6c32253f9826/cartography/intel/okta/users.py#L105 and I think it is a waste of a round trip to re-query the graph.  

This comment also applies to querying users [here](https://github.com/lyft/cartography/blob/952620fdf3ff15c6d8a3872227ff6c32253f9826/cartography/intel/okta/roles.py#L153) and groups [here](https://github.com/lyft/cartography/blob/952620fdf3ff15c6d8a3872227ff6c32253f9826/cartography/intel/okta/roles.py#L162).

The solution I proposed before was to use our [`_attach()` pattern](https://github.com/lyft/cartography/blob/9da67c69389e1a6d4555a5efcfe3e3120fa26460/docs/writing-intel-modules.md#connecting-different-node-types-with-the-_attach-pattern).  However, this would be a significant refactor as we would need to move FactorClient around, and we would also need to do similar work for roles.  I recognize though that Okta's API around Factors does not list them, requiring us to provide a user ID (which is somewhat cumbersome for us).

I think we should remove all of the `get_*_from_graph()` functions because we already have this data in memory at various points in the code.  We should refactor to pass pointers to this data where it is needed and avoid incurring extraneous RTT hits.

I'll defer to others on the team if they decide that this isn't that big of an issue though.